llvm/lib/Target/AMDGPU/BUFInstructions.td

```diff
@@ -2895,28 +2887,25 @@ multiclass MTBUF_Real_AllAddr_gfx11_gfx12_Renamed_Impl<bits<4> op, string real_n
   defm _VBUFFER_OFFSET : VBUFFER_MTBUF_Real_gfx12<op, real_name>;
 }
 
-multiclass MTBUF_Real_AllAddr_gfx11_gfx12<bits<4> op>
- : MTBUF_Real_AllAddr_gfx11_gfx12_Renamed_Impl<op, get_BUF_ps<NAME>.Mnemonic>;
-
-
-multiclass MTBUF_Real_AllAddr_gfx11_gfx12_Renamed<bits<4> op, string real_name>
-  : MTBUF_Real_AllAddr_gfx11_gfx12_Renamed_Impl<op, real_name> {
+multiclass MTBUF_Real_AllAddr_gfx11_gfx12<bits<4> op, string real_name = !tolower(NAME)>
```

Comment:
```suggestion
multiclass MTBUF_Real_AllAddr_gfx11_gfx12<bits<4> op, string real_name = get_BUF_ps<NAME>.Mnemonic
```

It is assumed now that !tolower(NAME) == !tolower(NAME), but I think there is no need to make that assumption. Using the suggested formulation avoids it.

Reply:
True - but I'm not aware of any cases where `!tolower(NAME) != get_BUF_ps<NAME>.Mnemonic`, so I'm not sure if it's a useful abstraction.

Reply:
I've tried to clean this up in #84524. See what you think.